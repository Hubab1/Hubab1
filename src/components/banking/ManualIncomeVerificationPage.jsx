import React from 'react';
import styled from '@emotion/styled';

import { BackButton } from 'components/common/BackLink';
import ActionButton from 'components/common/ActionButton/ActionButton';
import Capsule from 'components/common/Capsule/Capsule';
import { H1, H3 } from 'assets/styles';
import finance from 'assets/images/finance.png';

const SkinnyH1 = styled(H1)`
    width: 70%;
`;

const SpacedH3 = styled(H3)`
    margin-top: 15px;
    margin-bottom: 30px;
`;

export default function ManualIncomeVerificationPage (props) {
    return (
        <>
            <SkinnyH1>Income and Asset Verification</SkinnyH1>
            <SpacedH3>Add at least one income source or asset below.</SpacedH3>
            <Capsule
                prefix={<img alt="coin" src={finance}></img>}
                label="Income"
                buttonLabel="Add an Income Source"
                tip="TBD"
            />
            <ActionButton marginTop={40} marginBottom={20}>
                Continue
            </ActionButton>
            <BackButton onClick={props.goBack} />
        </>
    );
}
