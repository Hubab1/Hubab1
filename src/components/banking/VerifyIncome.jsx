import React, { Fragment } from 'react';
import styled from '@emotion/styled';
import { css } from 'emotion';

import { styles } from 'assets/styles';
import GreenCheckIcon from 'components/common/GreenCheckIcon';
import safeImage from 'assets/images/connect-bank/safe.png';
import padlockImage from 'assets/images/connect-bank/padlock.png';
import thunderImage from 'assets/images/connect-bank/thunder.jpg';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { H1, H3, P, Bold } from 'assets/styles';
import GenericFormMessage from 'components/common/GenericFormMessage';
import { ROUTES } from 'app/constants';
import { Link } from 'react-router-dom';

const SpacedH3 = styled(H3)`
    margin: 20px 5% 25px 5%;
`

const bodyContainer = css`
    margin: 40px 10px 40px 30px;
    text-align: left;
    display: flex;
    flex-direction: column;
    p{ margin-left: 15px;}
`

const iconColumn = css`
    margin-top: 5px;
`

const bodyRow = css`
    display: flex;
    margin-bottom: 20px;
`

const VerifyIncome = props => {
    return (
        <Fragment>
            <H1>Verify Your Income Instantly</H1>
            <SpacedH3>Confirm that you qualify for this apartment by linking your bank account.</SpacedH3>
            <img src={safeImage} alt="vault"/>
            <div className={bodyContainer}>
                <div className={bodyRow}>
                    <img src={padlockImage} alt="padlock" width="18" height="28" className={iconColumn} />
                    <P><Bold>Your data is safe.</Bold> Bank level encryption is used to connect and your credentials are never stored.</P>
                </div>
                <div className={bodyRow}>
                    <GreenCheckIcon/>
                    <P><Bold>Your money is safe.</Bold> This does not authorize any transactions from your account.</P>
                </div>
                <div className={bodyRow}>
                    <img src={thunderImage} alt="thunder" width="18" height="28" className={iconColumn} />
                    <P><Bold>Link your bank account for instant verification.</Bold> If your income is directly deposited to your bank account.</P>
                </div>
            </div>
            {!!props.errors && <GenericFormMessage type="error" messages={props.errors}/>}
            <ActionButton disabled={props.loadingFinicityIframe} onClick={props.openFinicityIframe} marginBottom={20}>
                Link Bank Account
            </ActionButton>
            <Link to={ROUTES.INCOME_VERIFICATION_SUMMARY} className={styles.linkNoStyle}>
                <ActionButton variant="outlined" marginBottom={20}>
                    Verify Manually
                </ActionButton>
            </Link>
        </Fragment>
    );
}


export default VerifyIncome;
