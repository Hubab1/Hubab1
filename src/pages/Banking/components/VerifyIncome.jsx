import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from '@emotion/styled';
import { css } from 'emotion';

import { ROUTES } from 'constants/constants';
import GreenCheckIcon from 'common-components/GreenCheckIcon/GreenCheckIcon';
import ActionButton from 'common-components/ActionButton/ActionButton';
import GenericFormMessage from 'common-components/GenericFormMessage/GenericFormMessage';
import { styles } from 'assets/styles';
import { H1, H3, P, Bold } from 'assets/styles';
import safeImage from 'assets/images/connect-bank/safe.png';
import padlockImage from 'assets/images/connect-bank/padlock.png';
import thunderImage from 'assets/images/connect-bank/thunder.jpg';
import { applicationPath } from 'reducers/renter-profile';

const SpacedH3 = styled(H3)`
    margin: 20px 5% 25px 5%;
`;

const bodyContainer = css`
    margin: 40px 10px 40px 30px;
    text-align: left;
    display: flex;
    flex-direction: column;
    p {
        margin-left: 15px;
    }
`;

const iconColumn = css`
    margin-top: 5px;
`;

const bodyRow = css`
    display: flex;
    margin-bottom: 20px;
`;

const VerifyIncome = ({ errors, loadingFinicityIframe, openFinicityIframe, reportNoIncomeAssets, application }) => {
    return (
        <>
            <H1>Verify Your Income and Assets</H1>
            <SpacedH3>
                Verifying your income and/or assets will help us ensure that you qualify for this apartment.
            </SpacedH3>
            <img src={safeImage} alt="vault" />
            <div className={bodyContainer}>
                <div className={bodyRow}>
                    <img src={thunderImage} alt="thunder" width="18" height="28" className={iconColumn} />
                    <P>
                        <Bold>Link your bank account for instant verification.</Bold> This option works best if your
                        income is directly deposited to your bank account.
                    </P>
                </div>
                <div className={bodyRow}>
                    <img src={padlockImage} alt="padlock" width="18" height="28" className={iconColumn} />
                    <P>
                        <Bold>Your data is safe.</Bold> Bank level encryption is used to connect and your credentials
                        are never stored.
                    </P>
                </div>
                <div className={bodyRow}>
                    <GreenCheckIcon />
                    <P>
                        <Bold>Your money is safe.</Bold> This does not authorize any transactions from your account.
                    </P>
                </div>
            </div>
            {!!errors && <GenericFormMessage type="error" messages={errors} />}
            <ActionButton disabled={loadingFinicityIframe} onClick={openFinicityIframe} marginBottom={20}>
                Link Bank Account
            </ActionButton>
            <Link
                to={applicationPath(ROUTES.INCOME_VERIFICATION_SUMMARY, application.id)}
                className={styles.linkNoStyle}
            >
                <ActionButton variant="outlined" marginBottom={20}>
                    Iâ€™d Rather Manually Add Income/Assets
                </ActionButton>
            </Link>
            <Link
                to={applicationPath(ROUTES.INCOME_VERIFICATION_SUMMARY, application.id)}
                className={styles.linkNoStyle}
                onClick={(e) =>
                    reportNoIncomeAssets(e, applicationPath(ROUTES.INCOME_VERIFICATION_SUMMARY, application.id))
                }
            >
                <ActionButton variant="outlined" marginBottom={20}>
                    I Don&apos;t Have Income or Assets
                </ActionButton>
            </Link>
        </>
    );
};

VerifyIncome.propTypes = {
    errors: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
    loadingFinicityIframe: PropTypes.bool,
    openFinicityIframe: PropTypes.func,
    reportNoIncomeAssets: PropTypes.func,
    application: PropTypes.object.isRequired,
};

export default VerifyIncome;
