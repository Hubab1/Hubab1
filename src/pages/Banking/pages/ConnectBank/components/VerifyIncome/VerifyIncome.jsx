import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from '@emotion/styled';
import { css } from 'emotion';

import { ROUTES } from 'constants/constants';
import GreenCheckIcon from 'components//GreenCheckIcon/GreenCheckIcon';
import ActionButton from 'components//ActionButton/ActionButton';
import GenericFormMessage from 'components//GenericFormMessage/GenericFormMessage';
import BankingContext from 'pages/Banking/BankingContext';
import { styles } from 'assets/styles';
import { H1, H3, P, Bold } from 'assets/styles';
import safeImage from 'assets/images/connect-bank/safe.png';
import padlockImage from 'assets/images/connect-bank/padlock.png';
import thunderImage from 'assets/images/connect-bank/thunder.jpg';

const SpacedH3 = styled(H3)`
    margin: 20px 5% 25px 5%;
`;

const bodyContainer = css`
    margin: 40px 10px 40px 30px;
    text-align: left;
    display: flex;
    flex-direction: column;
    p {
        margin-left: 15px;
    }
`;

const iconColumn = css`
    margin-top: 5px;
`;

const bodyRow = css`
    display: flex;
    margin-bottom: 20px;
`;

const VerifyIncome = ({ errors, loadingFinicityIframe, openFinicityIframe, reportNoIncomeAssets }) => {
    const context = React.useContext(BankingContext);
    if (!context.routeSelected) {
        return null;
    }

    return (
        <>
            <H1>Verify Your Income and Assets</H1>
            <SpacedH3>
                Verifying your income and/or assets will help us ensure that you qualify for this apartment.
            </SpacedH3>
            <img src={safeImage} alt="vault" />
            <div className={bodyContainer}>
                <div className={bodyRow}>
                    <img src={thunderImage} alt="thunder" width="18" height="28" className={iconColumn} />
                    <P>
                        <Bold>Link your bank account for instant verification.</Bold> This option works best if your
                        income is directly deposited to your bank account.
                    </P>
                </div>
                <div className={bodyRow}>
                    <img src={padlockImage} alt="padlock" width="18" height="28" className={iconColumn} />
                    <P>
                        <Bold>Your data is safe.</Bold> Bank level encryption is used to connect and your credentials
                        are never stored.
                    </P>
                </div>
                <div className={bodyRow}>
                    <GreenCheckIcon />
                    <P>
                        <Bold>Your money is safe.</Bold> This does not authorize any transactions from your account.
                    </P>
                </div>
            </div>
            {!!errors && <GenericFormMessage type="error" messages={errors} />}
            <ActionButton disabled={loadingFinicityIframe} onClick={openFinicityIframe} marginBottom={20}>
                Link Bank Account
            </ActionButton>
            <Link to={ROUTES.INCOME_VERIFICATION_SUMMARY} className={styles.linkNoStyle}>
                <ActionButton variant="outlined" marginBottom={20}>
                    Iâ€™d Rather Manually Add Income/Assets
                </ActionButton>
            </Link>
            <Link
                to={ROUTES.INCOME_VERIFICATION_SUMMARY}
                className={styles.linkNoStyle}
                onClick={(e) => reportNoIncomeAssets(e, ROUTES.INCOME_VERIFICATION_SUMMARY)}
            >
                <ActionButton variant="outlined" marginBottom={20}>
                    I Don&apos;t Have Income or Assets
                </ActionButton>
            </Link>
        </>
    );
};

VerifyIncome.propTypes = {
    errors: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
    loadingFinicityIframe: PropTypes.bool,
    openFinicityIframe: PropTypes.func,
    reportNoIncomeAssets: PropTypes.func,
};

export default VerifyIncome;
