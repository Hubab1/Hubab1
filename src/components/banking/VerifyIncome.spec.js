import React from 'react';
import { shallow } from 'enzyme';
import VerifyIncome from './VerifyIncome';
import ManualIncomeVerificationPage from './ManualIncomeVerificationPage';

let defaultProps;
beforeEach(() => {
    defaultProps = {
        goBack: jest.fn()
    }
})

it('shows manual income verification screen if screen === manual', () => {
    const wrapper = shallow(<VerifyIncome {...defaultProps} screen="manual"/>);
    expect(wrapper.find(ManualIncomeVerificationPage).length).toBe(1);
});
