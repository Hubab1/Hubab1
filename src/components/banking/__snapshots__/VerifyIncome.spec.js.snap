// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<React.Fragment>
  <ForwardRef(render)>
    Verify Your Income Instantly
  </ForwardRef(render)>
  <ForwardRef(render)>
    Confirm that you qualify for this apartment by linking your bank account.
  </ForwardRef(render)>
  <img
    alt="vault"
    src="safe.png"
  />
  <div
    className="css-1qmtr65"
  >
    <div
      className="css-1xu96tg"
    >
      <img
        alt="padlock"
        height="28"
        src="padlock.png"
        width="18"
      />
      <ForwardRef(render)>
        <ForwardRef(render)>
          Your data is safe.
        </ForwardRef(render)>
         Bank level encryption is used to connect and your credentials are never stored.
      </ForwardRef(render)>
    </div>
    <div
      className="css-1xu96tg"
    >
      <GreenCheckIcon />
      <ForwardRef(render)>
        <ForwardRef(render)>
          Your money is safe.
        </ForwardRef(render)>
         This does not authorize any transactions from your account.
      </ForwardRef(render)>
    </div>
  </div>
  <GenericFormDetail
    messages="Some random error"
    type="error"
  />
  <ActionButton
    color="primary"
    marginBottom={20}
    successGreen={false}
    variant="contained"
  >
    Link Bank Account
  </ActionButton>
  <ForwardRef
    className="css-1qj8w5r"
    to="/income-employment/summary"
  >
    <ActionButton
      color="primary"
      marginBottom={20}
      successGreen={false}
      variant="outlined"
    >
      Verify Manually
    </ActionButton>
  </ForwardRef>
</React.Fragment>
`;
